### PR TITLE
[core] Add new async helpers

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

### CODE REVIEW

**Review by Xaver-DaRed — APPROVED** [review]

**Review by claywar — APPROVED** [review]

**Comment by zach2good** [comment]
TODO: This will need some docs and some anonymous namespace stuff, because since we have `sql` exposed everywhere, and you can easily try and submit a job that has `_sql` or `sql` as an argument, you may or may not shadow or typo and use the sql connection on a different thread - ruining the lives of everyone around you.

**Comment by zach2good** [comment]
Thus the conversation we had the other day of wrapping all global uses of sql and lua as `global::sql` and `global::lua`.
It's a bit wordy, but it makes everything safe
